hibernate-core/src/main/java/org/hibernate/engine/spi/ActionQueue.java

```diff
@@ -281,6 +281,18 @@ private void addResolvedEntityInsertAction(AbstractEntityInsertAction insert) {
 			executeInserts();
 			LOG.debug( "Executing identity-insert immediately" );
 			execute( insert );
+
+			addAction(	// HHH-15258
+					OrphanRemovalAction.class,
+					new OrphanRemovalAction( insert.getId(),
+											 insert.getState(),
+											 null,
+											 insert.getInstance(),
+											 insert.getPersister(),
+											 false,
+											 (SessionImplementor) insert.getSession()
+					)
+			);
```

Comment:
The above bizarre action addition is only to ensure the query spaces are added, so the insertion event can be flushed.
Not elegant solution, but it seems ActionQueue related stuff is not elegant per se and other hack related to OrphanRemoval exists in the class as well.